packages/repository/src/relations/has-many/has-many-through-repository.factory.ts

```diff
@@ -65,7 +65,7 @@ export function createHasManyThroughRepositoryFactory<
     }
 
     function getThroughFkConstraint(
-      targetInstance: Target,
+      targetInstance: Partial<Target>,
```

Comment:
Interesting. If are accepting `Partial<Target>`, then how can we be sure that the partial object contains the properties we need to build the constraint? What if `targetInstance` is just `{title: 'Pencil'}`? (See #5592.)

Should we perhaps change this function to accept the key only?